src/Platform/Microsoft.Testing.Platform/Helpers/System/ISemaphore.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.Helpers;
+
+internal interface ISemaphore : IDisposable
```

Comment:
Ditto

Reply:
I would argue virtual dispatch cost is insignificant, after all we're dealing with I/O when logging which should be a few orders of magnitude slower. I know they are well tested in the runtime, but it's easier and more efficient to test all code paths by replacing them with mocks rather than wait for the semaphore implementation to timeout, for example.